src/python/pants/backend/docker/target_types.py

```diff
@@ -140,6 +149,46 @@ class DockerSkipPushField(BoolField):
     help = "If set to true, do not push this image to registries when running `./pants publish`."
 
 
+class DockerBuildOptionFieldMixin(ABC):
+    """Inherit this mixin class to provide options to `docker build`."""
+
+    docker_build_option: ClassVar[str]
+
+    @abstractmethod
+    def option_values(self) -> Iterator[str]:
+        """Subclasses must implement this, to turn their `self.value` into none, one or more option
+        values."""
+
+    @final
+    def options(self) -> Iterator[str]:
+        for value in self.option_values():
+            yield f"{self.docker_build_option}={shlex.quote(value)}"
+
+
+class DockerBuildSecretsOptionField(
+    AsyncFieldMixin, DockerBuildOptionFieldMixin, DictStringToStringField
+):
+    alias = "secrets"
+    help = "Secret file to expose to the build (only if BuildKit enabled).\n\n" + dedent(
```

Comment:
this could use more instructions, explaining that the key is the ID of the password and the value is the path to the secrets file. The path can either be an absolute path or a path relative to the BUILD file. Might be helpful to show both formats in the help string.

How do you decide on the ID of the password?

What should the file format of the secrets file be? Can you link to any references?